src/Equations/damaged-elastic/Kernels/Time.cpp

```diff
@@ -703,3 +700,1491 @@ void seissol::kernels::Time::flopsTaylorExpansion(long long& nonZeroFlops,
+  const real epsInitxy = m_damagedElasticParameters->epsInitxy;
+  const real epsInityz = m_damagedElasticParameters->epsInityz;
+  const real epsInitzx = m_damagedElasticParameters->epsInitzx;
+  const real damageParameter = data.material.local.Cd;
+  const real scalingValue = m_damagedElasticParameters->scalingValue;
+  const real breakCoefficient = scalingValue * damageParameter;
+  const real betaAlpha = m_damagedElasticParameters->betaAlpha;
+  const real aB0 = m_damagedElasticParameters->aB0;
+  const real aB1 = m_damagedElasticParameters->aB1;
+  const real aB2 = m_damagedElasticParameters->aB2;
+  const real aB3 = m_damagedElasticParameters->aB3;
+
+  for (unsigned int timeInterval = 0; timeInterval < CONVERGENCE_ORDER; ++timeInterval) {
+    real* exxNodal = (QInterpolatedBodyNodal[timeInterval] + 0 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* eyyNodal = (QInterpolatedBodyNodal[timeInterval] + 1 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* ezzNodal = (QInterpolatedBodyNodal[timeInterval] + 2 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* alphaNodal =
+        (QInterpolatedBodyNodal[timeInterval] + 9 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* breakNodal =
+        (QInterpolatedBodyNodal[timeInterval] + 10 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+
+    real* exyNodal = (QInterpolatedBodyNodal[timeInterval] + 3 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* eyzNodal = (QInterpolatedBodyNodal[timeInterval] + 4 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* ezxNodal = (QInterpolatedBodyNodal[timeInterval] + 5 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* vxNodal = (QInterpolatedBodyNodal[timeInterval] + 6 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* vyNodal = (QInterpolatedBodyNodal[timeInterval] + 7 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+    real* vzNodal = (QInterpolatedBodyNodal[timeInterval] + 8 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS);
+
+    real alpha_ave = alphaNodal[0];
+    real break_ave = breakNodal[0];
+    for (unsigned int q = 0; q < NUMBER_OF_ALIGNED_BASIS_FUNCTIONS - 1; ++q) {
+      break_ave = std::max(break_ave, breakNodal[q]);
+      alpha_ave = std::max(alpha_ave, alphaNodal[q]);
+    }
+
+    for (unsigned int q = 0; q < NUMBER_OF_ALIGNED_BASIS_FUNCTIONS; ++q) {
+      real EspI = (exxNodal[q] + epsInitxx) + (eyyNodal[q] + epsInityy) + (ezzNodal[q] + epsInitzz);
+      real EspII = (exxNodal[q] + epsInitxx) * (exxNodal[q] + epsInitxx) +
+                   (eyyNodal[q] + epsInityy) * (eyyNodal[q] + epsInityy) +
+                   (ezzNodal[q] + epsInitzz) * (ezzNodal[q] + epsInitzz) +
+                   2 * (exyNodal[q] + epsInitxy) * (exyNodal[q] + epsInitxy) +
+                   2 * (eyzNodal[q] + epsInityz) * (eyzNodal[q] + epsInityz) +
+                   2 * (ezxNodal[q] + epsInitzx) * (ezxNodal[q] + epsInitzx);
+      real xi;
+      if (EspII > 1e-30) {
+        xi = EspI / std::sqrt(EspII);
+      } else {
+        xi = 0.0;
+      }
```

Comment:
I see this code snippet everywhere duplicated. Please extract into a function `real ComputeXi(real EspI, EspII)`

Reply:
Done